File: scala/anagram/src/main/scala/anagram.scala
class Anagram(base_word: String) {
    def matches(words: Seq[String]): Seq[String] = {
        words.filter {
            word: String =>
                word.toLowerCase.sorted == base_word.toLowerCase.sorted &&
                    base_word.toLowerCase != word.toLowerCase
        }
    }
}

File: ecmascript/anagram/anagram.js
class Anagram {
  constructor(baseWord) {
    this.baseDict = this.wordDict(baseWord);
  }

  wordDict = (word) => {
    return [...word.toLowerCase()].sort().join('');
  }

  isAnagram = (word) => {
    return this.baseDict === this.wordDict(word);
  }
  matches(wordList, ...rest) {
    if(typeof(wordList) === 'string') {
      return [wordList, ...rest].filter(this.isAnagram);
    }
    return wordList.filter(this.isAnagram);
  }
}

export default Anagram;
